### ISSUE TITLE
[CT-3485] Semantic Layer custom calendar config support

### ISSUE THREAD

**Issue by QMalcolm** [dbt-labs/dbt-core#9265]
### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description
Details unknown.

The spec for this is still in the ideation phase.

### Acceptance criteria
People can specify the attributes necessary for custom calendar support 👀 

### Impact to Other Teams

* Semantic Layer - The semantic layer can't begin using this until we support it
* Cloud artifacts - This change will change the v12 manifest that goes out with 1.8

### Will backports be required?

Not sure if required, but the SL team would prefer a backport to 1.7

### Context

This issue should not be considered ready to work on until the work is done in dbt-semantic-interfaces, because the schema we need to implement isn't finalized until that point. Currently there does exist an issue for this work in dbt-semantic-interfaces.

**Comment by Jstein77** [comment]
@QMalcolm I wrote up our proposed implementation of this feature [here](https://www.notion.so/dbtlabs/Custom-Calendar-bb8e98131c1247b084dba242e05421db#6451e3fe4c994a778ef9554959bdaf8c). You can use this doc to help estimate how much work is needed for this ticket.

**Comment by apd-bbrewington** [comment]
Just to add some color on use case for this - could be super useful for companies/departments that rely heavily on a fiscal calendar

...which is typically available as table(s)/view(s) within the data warehouse

**Comment by FishtownBuildBot** [comment]
Opened a new issue in dbt-labs/docs.getdbt.com: https://github.com/dbt-labs/docs.getdbt.com/issues/6075
